src/types/onyx/Transaction.ts

```diff
@@ -548,6 +548,9 @@ type Transaction = OnyxCommon.OnyxValueWithOfflineFeedback<
 
         /** The inserted time of the transaction */
         inserted?: string;
+
+        /** Whether the merchant field was intentionally cleared by the user */
+        wasMerchantCleared?: boolean;
```

Comment:
I am still not sure if this is the right approach to add a flag. I'll check if this is an acceptable pattern.

Reply:
Is this what is pending?

Reply:
@mananjadhav I pinged you [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1751912181446779?thread_ts=1751830618.668549&cid=C01GTK53T8Q). If no one has any other suggestions, we should move forward with the solution

Reply:
I agree this is not the right solution. It's caused a few bugs now and we'd also need to keep search data in sync